x/interchainstaking/keeper/withdrawal_record.go

```diff
@@ -50,10 +50,14 @@
 	return sequence
 }
 
-func (k *Keeper) AddWithdrawalRecord(ctx sdk.Context, chainID, delegator string, distributions []*types.Distribution, recipient string, burnAmount sdk.Coin, hash string, status int32, completionTime time.Time, epochNumber int64) {
+func (k *Keeper) AddWithdrawalRecord(ctx sdk.Context, chainID, delegator string, distributions []*types.Distribution, recipient string, burnAmount sdk.Coin, hash string, status int32, completionTime time.Time, epochNumber int64) error {
 	record := types.WithdrawalRecord{ChainId: chainID, Delegator: delegator, Distribution: distributions, Recipient: recipient, Status: status, BurnAmount: burnAmount, Txhash: hash, CompletionTime: completionTime, EpochNumber: epochNumber}
+	if !record.BurnAmount.IsPositive() {
+		return fmt.Errorf("burnAmount cannot be negative or zero")
+	}
 	k.Logger(ctx).Info("addWithdrawalRecord", "record", record)
-	k.SetWithdrawalRecord(ctx, record)
+	err := k.SetWithdrawalRecord(ctx, record)
+	return err
```

Comment:
@joe-bowman - wdyt?

Reply:
We appear to be duplicating logic here. This will never fall through to SetWithdrawalRecord if burnAmount is non-positive, so we don't need to do the check in both places.